src/graph/objects/component.py

```diff
@@ -109,6 +116,18 @@ def needs(self):
         return self._needs
 
     async def validate(self, validate_specs=True):
+        """
+        Throws a `ValidationError` if the component is invalid. This call must be made
+        before any function with the `@required_validated` decorator is called.
+
+        args:
```

Comment:
Nit: It should be `Args` according to google style guide. Same to all the other docstrings.